subsys/bluetooth/audio/bap_unicast_client.c

```diff
@@ -727,17 +740,36 @@ static void unicast_client_ep_qos_state(struct bt_bap_ep *ep, struct net_buf_sim
 		LOG_ERR("No stream active for endpoint");
 		return;
 	}
+	ops = stream->ops;
 
-	if (ep->dir == BT_AUDIO_DIR_SINK && stream->ops != NULL && stream->ops->disabled != NULL) {
-		/* If the old state was enabling or streaming, then the sink
-		 * ASE has been disabled. Since the sink ASE does not have a
-		 * disabling state, we can check if by comparing the old_state
-		 */
-		const bool disabled = old_state == BT_BAP_EP_STATE_ENABLING ||
-				      old_state == BT_BAP_EP_STATE_STREAMING;
+	if (ops != NULL) {
+		if (ep->dir == BT_AUDIO_DIR_SINK && ops->disabled != NULL) {
+			/* If the old state was enabling or streaming, then the sink
+			 * ASE has been disabled. Since the sink ASE does not have a
+			 * disabling state, we can check if by comparing the old_state
+			 */
+			const bool disabled = old_state == BT_BAP_EP_STATE_ENABLING ||
+					      old_state == BT_BAP_EP_STATE_STREAMING;
+
+			if (disabled) {
+				ops->disabled(stream);
+			}
+		} else if (ep->dir == BT_AUDIO_DIR_SOURCE &&
+			   old_state == BT_BAP_EP_STATE_DISABLING && ops->stopped != NULL) {
+			/* We left the disabling state, let the upper layers know that the stream is
+			 * stopped
+			 */
+			uint8_t reason = ep->reason;
+
+			if (reason == BT_HCI_ERR_SUCCESS) {
+				/* Default to BT_HCI_ERR_UNSPECIFIED if no other reason is set */
+				reason = BT_HCI_ERR_UNSPECIFIED;
+			} else {
+				/* Reset reason */
+				ep->reason = BT_HCI_ERR_SUCCESS;
+			}
 
-		if (disabled) {
-			stream->ops->disabled(stream);
+			ops->stopped(stream, reason);
```

Comment:
if it is source  ASE state machine, stream should already been stopped before entering into "Disabling" state, why do we need to call stopped here ?